File: app/src/main/java/com/example/sunkai/heritage/Activity/UserCommentDetailActivity.java
package com.example.sunkai.heritage.Activity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.AsyncTask;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.util.Log;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import com.example.sunkai.heritage.ConnectWebService.HandleFind;
import com.example.sunkai.heritage.Data.FindActivityAllData;
import com.example.sunkai.heritage.Data.GlobalContext;
import com.example.sunkai.heritage.Data.HandlePic;
import com.example.sunkai.heritage.Data.CommentReplyData;
import com.example.sunkai.heritage.Data.UserCommentData;
import com.example.sunkai.heritage.R;
import com.example.sunkai.heritage.tools.MakeToast;

import java.io.ByteArrayInputStream;
import java.lang.ref.WeakReference;
import java.text.SimpleDateFormat;
import java.util.Collections;
import java.util.Date;
import java.util.List;
import java.util.ListIterator;

import static com.example.sunkai.heritage.Activity.LoginActivity.userID;

/**
 * 此类用于处理用户发帖详细信息页面
 */
public class UserCommentDetailActivity extends AppCompatActivity implements View.OnClickListener {
    public static final int ADD_COMMENT = 1;
    public static final int DELETE_COMMENT=2;
    private boolean isReply = false;

    private ImageView information_img;
    private TextView information_title;
    private TextView information_time;
    private TextView information_content;
    private TextView information_username;
    private TextView reverse;
    private TextView information_reply_num;
    private LinearLayout linearLayout4;
    private LinearLayout LinearLayout_reply;
    private EditText replyEdit;
    private Button replyBtn;
    private ProgressBar progressBar;
    UserCommentData data;
    List<CommentReplyData> datas;
    private ActionBar actionBack;

    private boolean isReverse = false;

    private static final String TAG = "UserCommentDetail";

    /**
     * 记录传入进来的帖子在原帖的位置和ID
     */
    private int commentID, inListPosition;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_comment_detail);
        initView();
        getData();

    }

    private void getData() {
        Bundle bundle = getIntent().getExtras();
        if (bundle != null && bundle.getSerializable("data") instanceof UserCommentData) {
            data = (UserCommentData) bundle.getSerializable("data");
            byte[] imageByte = getIntent().getByteArrayExtra("bitmap");
            Bitmap bitmap = HandlePic.handlePic(this, new ByteArrayInputStream(imageByte), 0);
            information_img.setImageBitmap(bitmap);
            commentID = data.getId();
            inListPosition = bundle.getInt("position");
            information_title.setText(data.getCommentTitle());
            information_time.setText(data.getCommentTime());
            information_content.setText(data.getCommentContent());
            information_reply_num.setText(data.getCommentReplyNum());
            setTitle(data.getUserName());
            new Thread(getReply).start();
        } else {
            int id = getIntent().getIntExtra("id", 0);
            Log.d(TAG, "onCreate: getID:" + id);
            if (id == 0)
                return;
            new getCommentInfo(id, this).execute();
        }
    }


    private void initView() {
        information_img = findViewById(R.id.information_img);
        information_title = findViewById(R.id.information_title);
        information_time = findViewById(R.id.information_time);
        information_content = findViewById(R.id.information_content);
        information_username = findViewById(R.id.information_username);
        information_reply_num = findViewById(R.id.information_reply_num);
        reverse = findViewById(R.id.user_comment_detail_reverse);
        linearLayout4 = findViewById(R.id.linearLayout4);
        LinearLayout_reply = findViewById(R.id.LinearLayout_reply);
        replyEdit = findViewById(R.id.reply_edittext);
        replyBtn = findViewById(R.id.reply_button);
        replyBtn.setOnClickListener(this);
        reverse.setOnClickListener(this);
        progressBar = findViewById(R.id.user_comment_detail_progressbar);
        actionBack = getSupportActionBar();
        if (actionBack != null) {
            actionBack.setDisplayHomeAsUpEnabled(true);
        }

    }


    //隐藏键盘
    private void hideKeyboard() {
        View view = getCurrentFocus();
        if (view != null) {
            InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
            if (inputMethodManager != null) {
                inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
            }
        }
    }

    //显示键盘
    private void showKeyboard() {
        View view = getCurrentFocus();
        if (view != null) {
            InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
            if (inputMethodManager != null) {
                inputMethodManager.showSoftInput(view, 1);
            }
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.reply_button:
                submit();
                break;
            case R.id.user_comment_detail_reverse:
                changeList();
                break;
        }
    }

    private void submit() {
        if (TextUtils.isEmpty(replyEdit.getText().toString().trim())) {
            Toast.makeText(UserCommentDetailActivity.this, "回复不能为空", Toast.LENGTH_SHORT).show();
            return;
        }
        if (userID == 0) {
            Toast.makeText(this, "没有登录", Toast.LENGTH_SHORT).show();
            Intent intent = new Intent(this, LoginActivity.class);
            intent.putExtra("isInto", 1);
            startActivityForResult(intent, 1);
            return;
        }
        String content = replyEdit.getText().toString();
        replyBtn.setVisibility(View.GONE);
        progressBar.setVisibility(View.VISIBLE);
        /*
         * 将回复的内容传给恢复类
         */
        HandleReply handleReply = new HandleReply(content);
        new Thread(handleReply.addReply).start();
    }

    private void changeList() {
        if (isReverse) {
            backupReverse();
            setTextViewBackup();
        } else {
            setReverse();
            setTextViewReverse();
        }
        isReverse = !isReverse;
    }

    private void setTextViewReverse() {
        reverse.setText(R.string.reverse_look);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            reverse.setTextColor(getColor(R.color.colorPrimary));
        } else {
            reverse.setTextColor(getResources().getColor(R.color.colorPrimary));
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            reverse.setCompoundDrawablesWithIntrinsicBounds(null, null, getDrawable(R.drawable.ic_arrow_upward_black_24dp), null);
        } else {
            reverse.setCompoundDrawablesWithIntrinsicBounds(null, null, getResources().getDrawable(R.drawable.ic_arrow_upward_black_24dp), null);
        }
    }

    private void setTextViewBackup() {
        reverse.setText(R.string.non_reverse_look);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            reverse.setTextColor(getColor(R.color.black));
        } else {
            reverse.setTextColor(getResources().getColor(R.color.black));
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            reverse.setCompoundDrawablesWithIntrinsicBounds(null, null, getDrawable(R.drawable.ic_arrow_downward_black_24dp), null);
        } else {
            reverse.setCompoundDrawablesWithIntrinsicBounds(null, null, getResources().getDrawable(R.drawable.ic_arrow_downward_black_24dp), null);
        }
    }

    private void setReverse() {
        LinearLayout_reply.removeAllViews();
        ListIterator<CommentReplyData> iterator = datas.listIterator(datas.size());
        hideKeyboard();
        for (; iterator.hasPrevious(); ) {
            CommentReplyData data = iterator.previous();
            setView(data);
        }
    }

    private void backupReverse() {
        LinearLayout_reply.removeAllViews();
        for (CommentReplyData data : datas) {
            setView(data);
        }
    }

    private void setView(CommentReplyData data) {
        LayoutInflater inflater = getLayoutInflater();
        @SuppressLint("InflateParams") View view = inflater.inflate(R.layout.user_comment_reply_information, null);
        Holder vh = new Holder();
        vh.name = view.findViewById(R.id.reply_name);
        vh.time = view.findViewById(R.id.reply_time);
        vh.content = view.findViewById(R.id.reply_content);
        vh.name.setText(data.getUserName());
        vh.time.setText(data.getReplyTime());
        vh.content.setText(data.getReplyContent());
        LinearLayout_reply.addView(view);
    }

    private void deleteComment(){
        new AlertDialog.Builder(this).setTitle("是否删除帖子?").setPositiveButton("删除", (dialog, which) -> {
            @SuppressLint("InflateParams")
            final AlertDialog ad= new AlertDialog.Builder(this).setView(LayoutInflater.from(this).inflate(R.layout.progress_view,null)).create();
            ad.show();
            new Thread(()->{
                final boolean result = HandleFind.Delete_User_Comment_By_ID(data.getId());
                runOnUiThread(()->{
                    if(ad.isShowing()){
                        ad.dismiss();
                    }
                    if(result){
                        MakeToast.MakeText(getResources().getString(R.string.delete_success));
                    }else{
                        MakeToast.MakeText(getResources().getString(R.string.has_problem));
                    }
                    setResult(DELETE_COMMENT,getIntent());
                    finish();
                });
            }).start();
        }).setNegativeButton("取消",((dialog, which) ->{} )).create().show();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                break;
            case R.id.user_comment_detail_item_delete:
                deleteComment();
        }
        return super.onOptionsItemSelected(item);
    }


    Runnable getReply = new Runnable() {
        @Override
        public void run() {
            List<CommentReplyData> getdatas = HandleFind.Get_User_Comment_Reply(data.getId());
            if (getdatas!=null) {
                datas = getdatas;
                runOnUiThread(()->{
                    hideKeyboard();
                    for (int i = 0; i < datas.size(); i++) {
                        CommentReplyData data = datas.get(i);
                        setView(data);
                    }
                });
            }
        }
    };

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        if (data != null && data.getUser_id() == userID) {
            getMenuInflater().inflate(R.menu.user_comment_detail_menu, menu);
        }
        return true;
    }

    static class getCommentInfo extends AsyncTask<Void, Void, FindActivityAllData> {
        private int id;
        private WeakReference<UserCommentDetailActivity> userCommentDetailWeakReference;

        getCommentInfo(int id, UserCommentDetailActivity userCommentDetail) {
            this.id = id;
            userCommentDetailWeakReference = new WeakReference<>(userCommentDetail);
        }

        @Override
        protected FindActivityAllData doInBackground(Void... voids) {

            return HandleFind.Get_All_User_Coment_Info_By_ID(LoginActivity.userID, id);
        }

        @Override
        protected void onPostExecute(FindActivityAllData findActivityAllData) {
            UserCommentDetailActivity userCommentDetail = userCommentDetailWeakReference.get();
            if (userCommentDetail == null || findActivityAllData == null)
                return;
            userCommentDetail.allDataSetView(findActivityAllData);
            new Thread(userCommentDetail.getReply).start();
        }
    }

    private void allDataSetView(FindActivityAllData data) {
        if(data.getImgCode()!=null) {
            byte[] imageByte = org.kobjects.base64.Base64.decode(data.getImgCode());
            Bitmap bitmap = HandlePic.handlePic(this, new ByteArrayInputStream(imageByte), 0);
            information_img.setImageBitmap(bitmap);
        }
        commentID = data.getId();
        information_title.setText(data.getComment_title());
        information_time.setText(data.getComent_time());
        information_content.setText(data.getComment_content());
        information_reply_num.setText(data.getReplyCount());
        setTitle(data.getUserName());
        //和老版本做一下兼容，复用代码
        this.data = new UserCommentData();
        this.data.setUserName(data.getUserName());
        this.data.setCommentReplyNum(data.getReplyCount());
        this.data.setId(data.getId());
        this.data.setCommentContent(data.getComment_content());
        this.data.setUser_id(data.getUserID());
        this.data.setCommentTime(data.getComent_time());
    }

    class HandleReply {
        int userID;
        String content, userName, replyTime;

        HandleReply(String content) {
            this.userID = LoginActivity.userID;
            this.content = content;
            this.userName = LoginActivity.userName;
        }

        Runnable addReply = new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(GlobalContext.Companion.getInstance(), UserCommentDetailActivity.class);
                intent.putExtra("id", commentID);
                String uriString = intent.toUri(Intent.URI_INTENT_SCHEME);
                Log.d(TAG, "uriString: " + uriString);
                final int result = HandleFind.Add_User_Comment_Reply(userID, commentID, content, uriString);
                SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
                replyTime = df.format(new Date());
                runOnUiThread(()->{
                    if(result>0) {
                        final CommentReplyData data=addDataToList(result);
                        setView(data);
                        resetWidge();
                        isReply = true;
                        MakeToast.MakeText("回复成功");
                        hideKeyboard();
                    }else{
                        MakeToast.MakeText("发生错误，请稍后再试");
                    }
                });
            }
        };
        private CommentReplyData addDataToList(int result){
            CommentReplyData data = new CommentReplyData();
            data.setReplyContent(content);
            data.setUserName(userName);
            data.setReplyTime(replyTime);
            data.setReplyId(result);
            datas.add(data);
            return data;
        }
        private void resetWidge(){
            progressBar.setVisibility(View.GONE);
            replyBtn.setVisibility(View.VISIBLE);
            information_reply_num.setText(String.valueOf(Integer.parseInt(information_reply_num.getText().toString()) + 1));
            replyEdit.setText("");
        }
    }

    class Holder {
        TextView name, time, content;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK && isReply) {
            Bundle bundle = new Bundle();
            bundle.putInt("commentID", commentID);
            bundle.putInt("position", inListPosition);
            Intent backIntent = new Intent();
            backIntent.putExtras(bundle);
            setResult(ADD_COMMENT, backIntent);
            finish();
        }

        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/com/example/sunkai/heritage/Activity/UserCommentDetailActivity.kt
package com.example.sunkai.heritage.Activity

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AlertDialog
import android.view.*
import android.view.animation.AnimationUtils
import androidx.core.transition.doOnEnd
import androidx.core.widget.toast
import com.bumptech.glide.Glide
import com.example.sunkai.heritage.Activity.BaseActivity.BaseHandleCollectActivity
import com.example.sunkai.heritage.Activity.LoginActivity.LoginActivity
import com.example.sunkai.heritage.Activity.LoginActivity.LoginActivity.Companion.userID
import com.example.sunkai.heritage.Adapter.UserCommentReplyRecyclerAdapter
import com.example.sunkai.heritage.ConnectWebService.BaseSetting
import com.example.sunkai.heritage.ConnectWebService.HandleFind
import com.example.sunkai.heritage.Data.CommentReplyInformation
import com.example.sunkai.heritage.Data.HandlePic
import com.example.sunkai.heritage.Data.UserCommentData
import com.example.sunkai.heritage.Dialog.AddUserCommentBottomDialog
import com.example.sunkai.heritage.Interface.AddUserReplyDialog
import com.example.sunkai.heritage.Interface.OnPageLoaded
import com.example.sunkai.heritage.R
import com.example.sunkai.heritage.tools.ThreadPool
import com.example.sunkai.heritage.value.*
import kotlinx.android.synthetic.main.activity_user_comment_detail.*

/**
 * 此类用于处理用户发帖详细信息页面
 */
class UserCommentDetailActivity : BaseHandleCollectActivity(), View.OnClickListener, OnPageLoaded {

    private var isReply = false
    internal var data: UserCommentData? = null

    private var isReverse = false

    //用于收藏的id
    private var commentID: Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_comment_detail)
        initView()
        getData()
        //从发现页点进来的时候执行动画，其他页面点进来不执行动画
        //当动画完成后，再显示帖子标题栏
        //防止在动画行进的时候，标题栏遮挡图片的问题
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && intent.getIntExtra(FROM, DEFAULT_FROM) != FROM_COLLECTION) {
            window.sharedElementEnterTransition.doOnEnd {
                getReplysInfo(commentID)
                showBackLinear()
            }
        } else {
            getReplysInfo(commentID)
            showBackLinear()
        }
    }

    private fun getData() {
        if (intent.getSerializableExtra(DATA) is UserCommentData) {
            data = intent.getSerializableExtra(DATA) as UserCommentData
            val data = data
            commentID = data?.id ?: 0
            data?.let {
                setUserCommentView(data, null)
            }

        } else {
            //从我的消息页面进入的时候，只会传来帖子id，这时候要获取全部的内容
            val id=intent.getIntExtra(ID,0)
            commentID=id
            ThreadPool.execute {
                if(LoginActivity.userID==0||id==0) finish()
                val userCommentData=HandleFind.GetAllUserCommentInfoByID(LoginActivity.userID,id)?:return@execute
                runOnUiThread {
                    setUserCommentView(userCommentData,null)
                    showBackLinear()
                    getReplysInfo(commentID)
                }
            }
        }
    }


    private fun initView() {
        userCommentAddReplyBtn.setOnClickListener(this)
        userCommentDetailReverse.setOnClickListener(this)
        setSupportActionBar(userCommentDetailToolbar)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
        userCommentSwipeRefresh.setOnRefreshListener {
            if (commentID != 0) {
                getReplysInfo(commentID)
            }
        }
    }

    private fun showBackLinear() {
        usercommentInformationLinear.visibility = View.VISIBLE
        val option = intent.getIntExtra(OPTION, COMMON_SHOW)
        if (option == ANIMATION_SHOW) {
            val animation = AnimationUtils.loadAnimation(this, R.anim.fade_in_quick)
            usercommentInformationLinear.startAnimation(animation)
        }
    }

    private fun changeList() {
        reverseData()
        if (isReverse) {
            setTextViewBackup()
        } else {
            setTextViewReverse()
        }
        isReverse = !isReverse
    }

    private fun setTextViewReverse() {
        userCommentDetailReverse.setText(R.string.reverse_look)
        userCommentDetailReverse.setTextColor(ContextCompat.getColor(this, R.color.colorPrimary))
        userCommentDetailReverse.setCompoundDrawablesWithIntrinsicBounds(null, null, ContextCompat.getDrawable(this, R.drawable.ic_arrow_upward_black_24dp), null)
    }

    private fun setTextViewBackup() {
        userCommentDetailReverse.setText(R.string.non_reverse_look)
        userCommentDetailReverse.setTextColor(ContextCompat.getColor(this, R.color.black))
        userCommentDetailReverse.setCompoundDrawablesWithIntrinsicBounds(null, null, ContextCompat.getDrawable(this, R.drawable.ic_arrow_downward_black_24dp), null)
    }

    private fun reverseData() {
        val adapter = userCommentReplyRecyclerView.adapter
        if (adapter is UserCommentReplyRecyclerAdapter) {
            adapter.reverseData()
        }
    }

    private fun setUserCommentView(data: UserCommentData, image: ByteArray?) {
        this.data=data
        informationTitle.text = data.commentTitle
        informationContent.text = data.commentContent
        informationReplyNum.text = data.replyNum.toString()
        title = data.userName
        if (image != null) {
            userCommentImage.setImageBitmap(HandlePic.handlePic(image))
        } else {
            Glide.with(this).load(BaseSetting.URL + data.imageUrl).into(userCommentImage)
        }
    }

    private fun deleteComment() {
        AlertDialog.Builder(this).setTitle("是否删除帖子?").setPositiveButton("删除") { _, _ ->
            val ad = AlertDialog.Builder(this).setView(LayoutInflater.from(this).inflate(R.layout.progress_view, userCommentReplyAppbar, false)).create()
            ad.show()
            ThreadPool.execute {
                val result = HandleFind.DeleteUserCommentByID(data!!.id)
                runOnUiThread {
                    if (ad.isShowing) {
                        ad.dismiss()
                    }
                    if (result) {
                        toast(resources.getString(R.string.delete_success))
                    } else {
                        toast(resources.getString(R.string.has_problem))
                    }
                    setResult(DELETE_COMMENT)
                    onBackPressed()
                }
            }
        }.setNegativeButton("取消") { _, _ -> }.create().show()
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            android.R.id.home -> onBackPressed()
            R.id.user_comment_detail_item_delete -> deleteComment()
            R.id.user_comment_detail_item_edit -> {
                val data=data
                data?.let {
                    val intent = Intent(this@UserCommentDetailActivity, ModifyUsercommentActivity::class.java)
                    intent.putExtra(DATA, data)
                    startActivityForResult(intent, UPDATE_USER_COMMENT)
                    return true
                }
            }
        }
        return super.onOptionsItemSelected(item)
    }

    //如果登陆用户是作者，则显示编辑和删除帖子的menu，否则，显示收藏menu
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val comemntData = data
        if (comemntData != null && comemntData.userID == userID) {
            menuInflater.inflate(R.menu.user_comment_detail_menu, menu)
            return true
        }
        return super.onCreateOptionsMenu(menu)
    }

    override fun getType(): String {
        return TYPE_FIND
    }

    override fun getID(): Int? {
        return if (commentID == 0) null else commentID
    }

    private fun getReplysInfo(commentID: Int) {
        if (commentID == 0) return
        onPreLoad()
        ThreadPool.execute {
            val datas = HandleFind.GetUserCommentReply(commentID)
            runOnUiThread {
                onPostLoad()
                val adapter = UserCommentReplyRecyclerAdapter(this, datas)
                userCommentReplyRecyclerView.adapter = adapter
            }
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK && isReply) {
            val bundle = Bundle()
            val backIntent = Intent()
            backIntent.putExtras(bundle)
            setResult(ADD_COMMENT, backIntent)
            finish()
        }
        return super.onKeyDown(keyCode, event)
    }

    private fun generateDialog(commentID: Int): AddUserCommentBottomDialog? {
        val dialog = AddUserCommentBottomDialog(this, commentID,data?:return null)
        dialog.setOnAddUserReplyListener(object : AddUserReplyDialog {
            override fun onAddUserReplySuccess(data: CommentReplyInformation) {
                val adapter = userCommentReplyRecyclerView.adapter
                if (adapter is UserCommentReplyRecyclerAdapter) {
                    adapter.addData(data)
                    userCommentReplyRecyclerView.smoothScrollToPosition(adapter.itemCount)
                    informationReplyNum.text = (informationReplyNum.text.toString().toInt() + 1).toString()
                }
            }
        })
        return dialog
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.userCommentDetailReverse -> changeList()
            R.id.userCommentAddReplyBtn -> generateDialog(commentID)?.show()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        //清除adapter的内容，防止返回时候的界面显示的问题
        this.commentID = 0
        userCommentReplyRecyclerView.adapter = null
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        when (requestCode) {
            UPDATE_USER_COMMENT -> {
                if (resultCode == UPDATE_SUCCESS) {
                    data?.let {
                        if (data.getSerializableExtra(DATA) is UserCommentData) {
                            setUserCommentView(data.getSerializableExtra(DATA) as UserCommentData, data.getByteArrayExtra("image"))
                        }
                    }
                }
            }
        }
    }

    override fun onPreLoad() {
        userCommentReplyRecyclerView.adapter = null
        userCommentSwipeRefresh.isRefreshing = true
    }

    override fun onPostLoad() {
        userCommentSwipeRefresh.isRefreshing = false
    }

    companion object {
        const val ADD_COMMENT = 1
        const val DELETE_COMMENT = 2
        const val COMMON_SHOW = 3
        const val ANIMATION_SHOW = 4
        const val FROM_COLLECTION = 0
        const val DEFAULT_FROM = -1
    }
}